src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -2519,33 +2523,20 @@ protected void VisitBinaryInterpolatedStringAddition(BoundBinaryOperator node)
             bool hasConditionalEvaluation = data.UsesBoolReturns || hasTrailingHandlerValidityParameter;
             TLocalState? shortCircuitState = hasConditionalEvaluation ? State.Clone() : default;
 
-            while (stack.TryPop(out var currentString))
+            foreach (var part in parts)
             {
-                visitedFirst |= VisitInterpolatedStringHandlerParts(currentString, data.UsesBoolReturns, firstPartIsConditional: visitedFirst || hasTrailingHandlerValidityParameter, ref shortCircuitState);
+                visitedFirst |= VisitInterpolatedStringHandlerParts(part, data.UsesBoolReturns, firstPartIsConditional: visitedFirst || hasTrailingHandlerValidityParameter, ref shortCircuitState);
             }
 
             if (hasConditionalEvaluation)
             {
                 Join(ref State, ref shortCircuitState);
             }
 
-            stack.Free();
+            parts.Free();
         }
 
-        protected virtual BoundBinaryOperator? PushBinaryOperatorInterpolatedStringChildren(BoundBinaryOperator node, ArrayBuilder<BoundInterpolatedString> stack)
-        {
-            stack.Push((BoundInterpolatedString)node.Right);
-            switch (node.Left)
-            {
-                case BoundBinaryOperator next:
-                    return next;
-                case BoundInterpolatedString @string:
-                    stack.Push(@string);
-                    return null;
-                default:
-                    throw ExceptionUtilities.UnexpectedValue(node.Left.Kind);
-            }
-        }
+        protected virtual void OnPushBinaryOperatorInterpolatedStringChildren(BoundBinaryOperator node) { }
```

Comment:
>OnPushBinaryOperatorInterpolatedStringChildren [](http://example.com/codeflow?start=31&length=46)

Given that the default implementation is empty, I have no idea what is the purpose of this method and why it has this name.

Reply:
I'll rename the method to be more clear.